internal/datasourcev2/types.go

```diff
@@ -0,0 +1,87 @@
+package datasourcev2
+
+import (
+	"encoding/json"
+
+	es "github.com/launchdarkly/eventsource"
+)
+
+type pollingPayload struct {
+	Events []event `json:"events"`
+}
+
+type event struct {
+	Name      string          `json:"name"`
+	EventData json.RawMessage `json:"data"`
+}
+
+// Begin es.Event interface implementation
+
+// Id returns the id of the event.
+func (e event) Id() string { //nolint:stylecheck // The interface requires this method.
+	return ""
+}
+
+// Event returns the name of the event.
+func (e event) Event() string {
+	return e.Name
+}
+
+// Data returns the raw data of the event.
+func (e event) Data() string {
+	return string(e.EventData)
+}
```

Comment:
Added these types so we could take the polling payload and treat it like a stream of events for processing in the functions I made in the other PR.